### PR TITLE
fix(kubernetes-templating): IPv6 support for Flux managed Cilium and …

### PR DESCRIPTION
**Pull request by samip5**
…CoreDNS

### CODE REVIEW

File: bootstrap/templates/kubernetes/apps/kube-system/cilium/app/helmrelease.yaml.j2 [hunk 1]
```diff
@@ -75,17 +76,29 @@ spec:
                 - *host
     ipam:
       mode: kubernetes
-    k8sServiceHost: "{{ bootstrap_kube_vip_addr }}"
+    {% if bootstrap_ipv6_enabled | default(false) %}
+    ipv4NativeRoutingCIDR: "{% raw %}{{ k3s_server['cluster-cidr'].split(',')[0] }}{% endraw %}"
+    ipv6NativeRoutingCIDR: "{% raw %}{{ k3s_server['cluster-cidr'].split(',')[1] }}{% endraw %}"
```
Comment (onedr0p) [inline bootstrap/templates/kubernetes/apps/kube-system/cilium/app/helmrelease.yaml.j2:81]:
```suggestion
    ipv4NativeRoutingCIDR: "{% raw %}{{ bootstrap_cluster_cidr.split(',')[0] }}{% endraw %}"
    ipv6NativeRoutingCIDR: "{% raw %}{{ bootstrap_cluster_cidr.split(',')[1] }}{% endraw %}"
```

Comment (onedr0p) [inline bootstrap/templates/kubernetes/apps/kube-system/cilium/app/helmrelease.yaml.j2:81]:
```suggestion
    ipv4NativeRoutingCIDR: "{% raw %}{{ bootstrap_cluster_cidr.split(',')[0] }}{% endraw %}"
    ipv6NativeRoutingCIDR: "{% raw %}{{ bootstrap_cluster_cidr.split(',')[1] }}{% endraw %}"
```

File: bootstrap/templates/kubernetes/apps/kube-system/cilium/app/helmrelease.yaml.j2 [hunk 2]
```diff
@@ -75,17 +76,29 @@ spec:
...
+    ipv6:
+      enabled: true
+    {% else %}
+    ipv4NativeRoutingCIDR: "{% raw %}{{ k3s_server['cluster-cidr'] }}{% endraw %}"
```
Comment (onedr0p) [inline bootstrap/templates/kubernetes/apps/kube-system/cilium/app/helmrelease.yaml.j2:85]:
```suggestion
    ipv4NativeRoutingCIDR: "{{ bootstrap_cluster_cidr }}"
```

File: bootstrap/templates/kubernetes/apps/kube-system/cilium/app/helmrelease.yaml.j2 [hunk 3]
```diff
@@ -75,17 +76,29 @@ spec:
...
+    {% endif %}
+    k8sServiceHost: "{% raw %}{{ kube_vip_addr }}{% endraw %}"
```
Comment (onedr0p) [inline bootstrap/templates/kubernetes/apps/kube-system/cilium/app/helmrelease.yaml.j2:87]:
```suggestion
    k8sServiceHost: "{{ bootstrap_kube_vip_addr }}"
```

File: bootstrap/templates/kubernetes/apps/kube-system/coredns/app/helmrelease.yaml.j2 [hunk 4]
```diff
@@ -31,7 +32,11 @@ spec:
     k8sAppLabelOverride: kube-dns
     service:
       name: kube-dns
-      clusterIP: "{{ bootstrap_coredns_addr }}"
+      {% if bootstrap_ipv6_enabled | default(false) %}
+      clusterIP: "{% raw %}{{ k3s_server['service-cidr'].split(',')[0] | ansible.utils.nthhost(10) }}{% endraw %}"
+      {% else %}
+      clusterIP: "{% raw %}{{ k3s_server['service-cidr'] | ansible.utils.nthhost(10) }}{% endraw %}"
+      {% endif %}
```
Comment (onedr0p) [inline bootstrap/templates/kubernetes/apps/kube-system/coredns/app/helmrelease.yaml.j2:39]:
```suggestion
      clusterIP: "{{ bootstrap_coredns_addr }}"
```

File: bootstrap/templates/kubernetes/apps/kube-system/cilium/app/kustomization.yaml.j2 [hunk 5]
```diff
@@ -1,7 +1,12 @@
...
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 namespace: kube-system
 resources:
+  {% if bootstrap_ipv6_enabled | default(false) %}
+  - ./helmrelease.yaml
+  {% else %}
   - ./cilium-l2.yaml
   - ./helmrelease.yaml
+  {% endif %}
```
Comment (onedr0p) [inline bootstrap/templates/kubernetes/apps/kube-system/cilium/app/kustomization.yaml.j2:12]:
```suggestion
  {% if not bootstrap_ipv6_enabled | default(false) %}
  - ./cilium-l2.yaml
  {% endif %}
  - ./helmrelease.yaml
```

**Review by onedr0p — CHANGES_REQUESTED** [review]

**Comment by samip5** [comment]
Now it does look right. :)

**Review by onedr0p — APPROVED** [review]

**Pull request merged**